boringtun/src/device/mod.rs

```diff
@@ -904,11 +904,12 @@ impl Device {
                 // The conn_handler handles packet received from a connected UDP socket, associated
                 // with a known peer, this saves us the hustle of finding the right peer. If another
                 // peer gets the same ip, it will be ignored until the socket does not expire.
-                let iface = &t.iface;
                 let mut iter = MAX_ITR;
 
                 let mut public_key = String::with_capacity(32);
-                for byte in peer.lock().tunnel.peer_static_public().as_bytes() {
+                let mut p = peer.lock();
```

Comment:
Won't this mean, that peer lock is held while receive [loop](https://github.com/NordSecurity/boringtun/pull/23/files#diff-9489a7524e0e996e901d2003e05a453a9b7095d37b296f403b023067b8a5563aR922) is running?

Reply:
Fixed